### ISSUE TITLE
Allow nullable type arguments in wildcard-typed array stores

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#1856]
With JSpecify experimental support enabled, NullAway rejects storing a value whose type argument is nullable into an array slot with an unbounded wildcard component type.

Minimal reproducer:

```java
import java.util.concurrent.CompletableFuture;

class Test {
  static void nullableArrayStore() {
    var futures = new CompletableFuture<?>[1];
    futures[0] = CompletableFuture.runAsync(() -> {});
  }
}
```

Actual diagnostic:

```
incompatible types: CompletableFuture<@Nullable Void> cannot be converted to CompletableFuture<capture of ?>
```

This should be accepted: `CompletableFuture<@Nullable Void>` is a subtype of `CompletableFuture<?>`. Assigning the same expression to a local variable of type `CompletableFuture<?>` is accepted:

```java
CompletableFuture<?> future = CompletableFuture.runAsync(() -> {});
```

Storing a concrete non-null instantiation in the same array is also accepted:

```java
static void concreteArrayStore(CompletableFuture<String> value) {
  var futures = new CompletableFuture<?>[1];
  futures[0] = value;
}
```

Configuration:

- `-XepOpt:NullAway:JSpecifyMode=true`
- `-XepOpt:NullAway:JSpecifyExperimental=true`

Originally reported as finding 3 in https://github.com/ben-manes/caffeine/issues/2004#issuecomment-5467003874.
